### PR TITLE
Port WebSocket from mm-redux and batch actions

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR ports the WebSocket implementation found in mattermost-redux to be used directly in the mobile app, and almost every event from the WebSocket is batching requests and actions to improve on performance.

While working on this I took the liberty to avoid some other unrelated actions from being dispatched when there was no need.

Also with this PR was introduced support for TypeScript and added some aliases to the imports

**Note for testing**: This code change is touching every single WebSocket event

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23238

### CODE REVIEW

File: app/actions/websocket.ts [hunk 1]
```diff
@@ -0,0 +1,1157 @@
...
+                    }
+                }
+            }
+
+            if (post.channel_id === currentChannelId) {
+                const id = post.channel_id + post.root_id;
+                const {typing} = state.entities;
+
+                if (typing[id]) {
+                    actions.push({
```
Comment (enahum) [inline app/actions/websocket.ts]: Maybe this should be a separate dispatch and then queue the other ones to be dispatch after a few milliseconds, that may help with the animation causing the post list jump. @migbot thoughts about that?

File: app/actions/device/index.js [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
-        dispatch(networkStatusChangedAction(isOnline));
-        dispatch({
-            type: DeviceTypes.CONNECTION_CHANGED,
-            data: isOnline,
-        });
+        if (isOnline !== undefined) { //eslint-disable-line no-undefined
+            dispatch(batchActions([
+                networkStatusChangedAction(isOnline), {
+                    type: DeviceTypes.CONNECTION_CHANGED,
+                    data: isOnline,
```
Comment (chuttam) [inline app/actions/device/index.js]: Recommend adding a name to batched actions.
(Leaving as single review comment for all `batchAction` calls in PR.)

File: app/actions/views/post.js [hunk 3]
```diff
@@ -292,42 +351,39 @@ function getPostsAdditionalDataBatch(posts = []) {
...
-        return actions;
+        return {data};
     };
 }
 
-function profilesStatusesAndToLoadFromPosts(posts = []) {
-    return (dispatch, getState) => {
-        const state = getState();
-        const {currentUserId, profiles, statuses} = state.entities.users;
+function profilesStatusesAndToLoadFromPosts(state, posts = []) {
```
Comment (chuttam) [inline app/actions/views/post.js]: This name has aways confused me. Perhaps, `userMetadataToLoadFromPosts`? I'm 0/5 on the proposed alternative, and 1/5 on renaming.

With consolidation/batching of functions we naturally are stuck with long function names with a lot of conjunctions (`and` this `and` that) in the name. I tend to prefer more explicit naming, even if the names get really long.

File: app/actions/device/index.js (hunk 2 above)
Comment (enahum) [reply app/actions/device/index.js]: DONE!!!! please take a look

Comment (chuttam) [reply app/actions/device/index.js]: Awesome. And the `BATCH_` prefix makes sense too. Also, easier to find in code searches. Thanks.

**Comment by enahum** [comment]
@josephbaylon that happened while adding a new DM from another client?

**Comment by josephbaylon** [comment]
> @josephbaylon that happened while adding a new DM from another client?

That's for existing ones. Also, when another user from webapp DMs the current user, the DM list is refreshed properly show the previous DMs that's supposed to be there. However, when the current user logs out and logs back in, the same issue of DM list being empty occurs.

**Comment by enahum** [comment]
Alright, I'll take a look to see why initial load is not working.. at least we know that the WS is fine

**Comment by enahum** [comment]
Ok @josephbaylon found the issue, thanks for testing... Will add some unit tests that I skipped last time and then I'll submit the fix

**Comment by enahum** [comment]
@josephbaylon that has nothing to do with the PR, the current `de` translation is master is not actually translated as you can see here https://github.com/mattermost/mattermost-mobile/blob/master/assets/base/i18n/de.json#L73 however on the release branch it will be translated https://github.com/mattermost/mattermost-mobile/blob/release-1.30/assets/base/i18n/de.json#L76

That is cause of the way translations are added to the branches

**Pull request merged**